internal/daemon/cache/repository_refresh_test.go

```diff
@@ -29,6 +25,14 @@ func noopRetrievalFn[T any](context.Context, string, string) ([]T, error) {
 	return nil, nil
 }
 
+// staticRetrievalFn returns a function that satisfies the With*RetrievalFn
+// and returns the provided slice and a nil error always
+func staticRetrievalFn[T any](ret []T) func(context.Context, string, string) ([]T, error) {
```

Comment:
static retrieval of what?  while this is clever, I'm left scratching my head.
perhaps a better name might help?  Hum... `testStaticResourceRetrievalFunc`???